ros/src/twist_controller/twist_controller.py

```diff
@@ -77,18 +83,26 @@ def control(self,
                           steer, cte, corrective_steer, predictive_steer)
 
             vel_delta = linear_velocity - current_velocity
-            accel = self.accel_pid.step(error=vel_delta, sample_time=sample_time)
+            res_accel = self.accel_pid.step(error=vel_delta, sample_time=sample_time)
+            
+            ###### Filter start
+            self.accel_t1 = np.delete(np.concatenate((self.accel_t1, np.ones(1) * res_accel)), 0)
```

Comment:
Gustavo. I am really struggling to understand what is going on here.
Is the whole idea to have a rolling window of values and then take sum of them periodically?
Why not just use collections.deque to make it obvious what is going on?
I am just not too smart and prefer simplifying everything to avoid any potential for bugs due to overcomplicating things.

Reply:
Agree, it will be changed when I find a way to reduce the delay but still keep gradual increasing.